src/deploy/functions/prepare.ts

```diff
@@ -43,6 +43,12 @@ export async function prepare(
       /* silent=*/ true
     ),
     ensureCloudBuildEnabled(projectId),
+    ensureApiEnabled.ensure(
```

Comment:
This will throw an error if it is unable to enable the API for some reason. We ought to catch this error so that this doesn't break users who can't use AR for some reason.

Reply:
using `then(()=>{}, ()=>{})` to catch errors. This will cause us to not see any error message if the enablement fails and the customer tries to dpeloy on GCFv2 though.

Reply:
Seems reasonable to me. To make our lives easier when debugging v2 issues, we could probably throw a logger.debug() in the catch. 
Alternatively, we could move this below `wantBackend` and actually throw the error if anything will be deployed on gcfv2.

Reply:
Doing that. Making it not silent either. By putting it below it should throw with a clearer error that halts deploys rather than failing the functions deploys but succeeding the deploy command overall.